openapi-cli/src/main/java/io/ballerina/openapi/cmd/BallerinaCodeGenerator.java

```diff
@@ -396,30 +395,37 @@ private List<GenSrcFile> generateClientFiles(Path openAPI, Filter filter, boolea
         licenseHeader = licenseHeader.isBlank() ? DO_NOT_MODIFY_FILE_HEADER : licenseHeader;
         TypeHandler.createInstance(openAPIDef, nullable);
         BallerinaClientGenerator clientGenerator = getBallerinaClientGenerator(oasClientConfig);
-        String mainContent = Formatter.format(clientGenerator.generateSyntaxTree()).toSourceCode();
+        SyntaxTree syntaxTree = clientGenerator.generateSyntaxTree();
         //Update type definition list with auth related type definitions
         List<TypeDefinitionNode> authNodes = clientGenerator.getBallerinaAuthConfigGenerator()
                 .getAuthRelatedTypeDefinitionNodes();
-        for (TypeDefinitionNode typeDef: authNodes) {
+        for (TypeDefinitionNode typeDef : authNodes) {
             TypeHandler.getInstance().addTypeDefinitionNode(typeDef.typeName().text(), typeDef);
         }
-        sourceFiles.add(new GenSrcFile(GenSrcFile.GenFileType.GEN_SRC, srcPackage, CLIENT_FILE_NAME,
-                licenseHeader + mainContent));
-        String utilContent = Formatter.format(
-                clientGenerator.getBallerinaUtilGenerator().generateUtilSyntaxTree()).toString();
-        if (!utilContent.isBlank()) {
-            sourceFiles.add(new GenSrcFile(GenSrcFile.GenFileType.UTIL_SRC, srcPackage, UTIL_FILE_NAME,
-                    licenseHeader + utilContent));
-        }
 
-        // Generate ballerina records to represent schemas.
-        SyntaxTree schemaSyntaxTree = TypeHandler.getInstance().generateTypeSyntaxTree();
-
-        String schemaContent = Formatter.format(schemaSyntaxTree).toSourceCode();
-        if (!schemaContent.isBlank()) {
-            sourceFiles.add(new GenSrcFile(GenSrcFile.GenFileType.MODEL_SRC, srcPackage, TYPE_FILE_NAME,
-                    licenseHeader + schemaContent));
+        if (singleFile) {
+            syntaxTree = SingleFileGenerator.combineSyntaxTrees(syntaxTree,
+                    clientGenerator.getBallerinaUtilGenerator().generateUtilSyntaxTree(),
+                    TypeHandler.getInstance().generateTypeSyntaxTree());
+        } else {
+            String mainContent = Formatter.format(syntaxTree).toSourceCode();
+            sourceFiles.add(new GenSrcFile(GenSrcFile.GenFileType.GEN_SRC, srcPackage, CLIENT_FILE_NAME,
+                    licenseHeader + mainContent));
+            String utilContent = Formatter.format(
+                    clientGenerator.getBallerinaUtilGenerator().generateUtilSyntaxTree()).toString();
```

Comment:
We could define a variable for this before the `if (singleFile)` check and reuse it in both places